### DISCUSSION TITLE
A bit confused by the various settings

### DISCUSSION THREAD

**Discussion by sygolene** [AlfMikael/snap-generator#13] [General]
Hello, 

First, I'd like to congratulate you and thank you for the fine tool you've designed. I was struggling for weeks with my own designs and finding this was kind of a revelation :-)...
That being said, even though I read the doc, I still struggle a bit to map the various settings to their effective implementation on the end model. 
So I went and did a drawing based on the default "Pin" (not the simple one, since I want to understand all the parameters). 
I could of course figure some settings, taking into account the various gaps, but can't quite map everything, still. 
Could you have a look to help fill in the gaps?
![Drawing](https://github.com/user-attachments/assets/142a884a-fe08-4e40-a5d4-bd4b88edc251)
![Settings](https://github.com/user-attachments/assets/15a8b5ff-d991-49f2-9771-be5eccf654c6)

Thanks in advance,

**Comment by AlfMikael** [comment]
Hey! Sorry I didn't see the notification until now. What I would suggest to you is to first set everything in the Gaps section to zero, and then look at the dimensions. Then it will probably make a lot more sense. After than you can change the different gap dimensions and see what happens.
I see that you are curious about a large number of dimensions, but rather than answer that I would just start by asking: does it matter? All those dimensions can be represented by equations with the different parameters, but the reason why I've not included it, is because it gets quite convoluted and I could not picture myself using it for anything when designing. If you're wondering "will the pin fit inside my parts?" then I would suggest looking at the **subtraction body**. It's geometry is simple, and the pin is always guaranteed to fit inside it.

Sorry for not answering your question completely, and if you're still confused, let me know :)

**Reply by sygolene** [reply]
Hello ! 

thanks for your return. While waiting, I sort of steered away from the "knowing it all" approach and basically played around with only three settings, thickness, size & nose angle with pretty nice results... I guess you're right! Still I think a nice and clean doc wouldn't go amiss as a good practice but I can imagine you'd rather work on features :D! 
In any event, I'm very thankful for this plugin and as of today, I don't really have any question yet so thanks :-)
